lib/types/map.js

```diff
@@ -119,7 +119,7 @@ class MongooseMap extends Map {
           v.$__.wasPopulated = { value: v._id };
           return v;
         });
-      } else {
+      } else if (value) { // Fix when using delete method because the value is undefined
```

Comment:
There is probably a better fix to do because of the value is `undefined`, we can probably skip a lot of checks